api/models/enrollment_tests.py

```diff
@@ -81,11 +83,34 @@ def __str__(self) -> str:
     @property
     def resulting_level(self) -> str:
         """Returns language level of the student based on amount of correct answers."""
-        correct_answers = self.answers.filter(is_correct=True).count()
-        total_answers = self.answers.count()
-        percentage = correct_answers / total_answers * 100
-        closest_percentage = min(
-            LEVEL_BY_PERCENTAGE_OF_CORRECT_ANSWERS, key=lambda x: abs(x - percentage)
-        )
+        return self.calculate_level([answer.id for answer in self.answers.all()])
+
+    @staticmethod
+    def calculate_level(answer_ids: Sequence[int]) -> str:
+        """Calculates language level depending on amount of correct answers.
+
+        This method can be called from outside without creating any `EnrollmentTestResult` objects.
+        """
+        # depending on number of questions in test, the thresholds are different
+        thresholds_for_number_of_questions = {
```

Comment:
Maybe let's save it as a constant outside the function?

Reply:
Done